### PR TITLE
Feature/rcb 3 create login page for admins and cooks

### PR DESCRIPTION
**Pull request by AlinFlorescu80**

### CODE REVIEW

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { ReactComponent as Logo } from "../../assets/images/recipe_box_logo_dark.svg";
+import { ReactComponent as CloseButton } from "../../assets/icons/close.svg";
+import "./UserLogin.css";
+import { backendUrl } from "../../App";
+
+export default function UserLogin() {
+    const navigate = useNavigate();
+    const [errors, setErrors] = useState<{
+        username: boolean;
+        password: boolean;
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx]: here you can just assign false to username and password directly

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                },
+                body: JSON.stringify({ username, password }),
+            });
+
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+
+            const data = await response.json();
+            navigate("/recipeslist");
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx:48]: you can try to store the data in local or session storage

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                </div>
+                <input
+                    type="text"
+                    ref={usernameRef}
+                    className={`input-with-icon ${
+                        errors.username ? "input-error" : ""
+                    }`}
+                />
+                <div className="validation-message">
+                    You must enter your username
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx]: Render the errors only if there is an error

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                <input
+                    type="password"
+                    ref={passwordRef}
+                    className={`input-with-icon ${
+                        errors.password ? "input-error" : ""
+                    }`}
+                />
+                <div className="validation-message">
+                    You must enter your password
+                </div>
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.tsx]: Same here, add a condition for rendering the error

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css [hunk 5]
```diff
@@ -0,0 +1,110 @@
+.login-card {
+    width: 460px;
+    height: 715px;
+    padding: 50px;
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css:4]: I suggest making the login card smaller

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+.login-logo {
+    margin: 100px auto 65px;
+    width: 350px;
+    height: auto;
+}
+.input-group {
+    width: 100%;
+    margin-bottom: 30px;
+    display: flex;
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css:37]: The space(margin) between inputs and buttons is too big

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    padding-left: 50px;
+    background-position: 5px center;
+}
+
+.login-button {
+    padding: 10px;
+    height: 45px;
+    background-color: #01ae3a;
+    color: white;
+    margin-top: 50px;
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css]: same here, the margin is too big

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+.input-group input[type="text"] {
+    background: white url(../../assets/icons/person_icon.png) left no-repeat;
+    background-size: 2rem 1.5rem;
+    padding-left: 50px;
+    background-position: 3px center;
+}
+.input-group input[type="password"] {
+    font-size: 2.5rem;
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css:71]: this font size is bigger than the one for the username, try making it consistent

File: recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css (hunk 6 above)
Comment (AlinFlorescu80) [reply recipe-box-frontend/src/main/frontend/src/Pages/LoginPage/UserLogin.css:37]: Screenshot 2024-09-06 185028

But here I used 50 and it looks about right

(I don't know if I pasted the image correctly)

**Review by en-catrinel-lupu — APPROVED** [review]

**Pull request merged**